### ISSUE TITLE
Incompatible with IE11

### ISSUE THREAD

**Issue by charlesroper** [FieldStudiesCouncil/tombiovis#82]
Had a support email through from someone reporting an Identikit would not load on their system (they get the text at the top but the JS doesn't load). On further investigation, it turned out they are using IE11 - which does indeed seem to be incompatible (more details on that below).

![2019-09-19_11h03_54](https://user-images.githubusercontent.com/176/65234819-37c02180-dacd-11e9-8557-e8bb1a01cc38.png)

Can we include a [JS feature test](https://gomakethings.com/conditionally-loading-javascript-only-when-the-browser-supports-it/) to detect if the browser supports the required code then present a message to the user if not? Or, if possible, use a polyfill ([polyfill.io is great for this](https://polyfill.io/v3/))? IE11 is still a relatively popular browser (between 4% and 9% marketshare, depending on where you get your stats from), so it needs at least some sort of graceful degradation in place to avoid confusing users.

I've tried loading a few sites with IE11 and all fail in the same way. Here are a couple of screenshots from IE11 devtools when accessing the test kit: https://fieldstudiescouncil.github.io/tombiovis/identikit/vis.html

![2019-09-19_10h52_54](https://user-images.githubusercontent.com/176/65234156-f8450580-dacb-11e9-8a33-cfde750bde75.png)
![2019-09-19_10h53_05](https://user-images.githubusercontent.com/176/65234166-faa75f80-dacb-11e9-9fa0-2b20c57440d9.png)

**Comment by burkmarr** [comment]
Yes I was aware that it doesn't work on IE11 (Edge is okay) and agree we should incorporate a JS feature or browser test and a message to warn people of it. I've added a bug tag to this issue so that we pick it up when we next work on it. In the meantime, we could add a message to the readme file about browser compatibility.

I don't think it's worth doing a lot of work to accommodate legacy browsers - I might take a different view if we were supporting a commercial product and/or were funded to maintain it.

**Comment by charlesroper** [comment]
Yes, it seems a tall order to make it compatible with IE11. Polyfill.io can be a quick win, but it's not a sure thing. The confusing thing for users is that it fails without warning, so a simple message on the page would suffice, I think. This message could be generated by doing a simple JS feature test. E.g., we could check for service worker compatibility (even if we're not using SWs), which would pass on modern browsers like Firefox, Edge and Chrome, but would fail on IE11; so we could do something like this:

```js
if ('serviceWorker' in navigator) {
    // load Identikit JS
} else {
   document.write('<p>Identikit requires a modern browser like Firefox, Edge, or Chrome...</p>')
}
```

See here for what browsers would pass/fail this particular test: https://caniuse.com/#search=serviceworker
